pkg/synthetictests/duplicated_events.go

```diff
@@ -126,6 +126,7 @@ var allowedRepeatedEventPatterns = []*regexp.Regexp{
 
 var allowedRepeatedEventFns = []isRepeatedEventOKFunc{
 	isConsoleReadinessDuringInstallation,
+	isConfigOperatorReadinessFailed,
```

Comment:
Just curious, it looks like isConsoleReadinessDuringInstallation is saying that event is ok during that time so skip it. But isConfigOperatorReadinessFailed is saying 'yes(true)' it failed are we wanting allowedRepeatedEvents to skip it in that case?

Reply:
The functions in the `[]isRepeatedEventOKFunc` are functions that return true if a particular message is to allowed; it does this by using functions because comparing just a pattern may not be enough (as done in the `[]allowedRepeatedEventPatterns` patterns); in my case, I needed the pattern to match but also needed the namespace and pod name to match.

In both cases, we are not checking for failure or being ok -- but just answering the question: "is this message allowed"?  The point of checking if a message is allowed is to avoid failing on that message twice: once when processing all the events and once when you write a specific test for it.  Since a separate test was introduced for the config-operator messages, we want to allow it here.

Reply:
Thank you, that helps me understand the logic and that I might need something similar for [trt-579](https://github.com/openshift/origin/pull/27454)